napari/_qt/qt_main_window.py

```diff
@@ -440,6 +452,131 @@ def __init__(self, viewer, *, show: bool = True):
         if show:
             self.show()
 
+    def _add_plugins_theme(self, event=None):
+        """Update stylesheet/icon/theme."""
+        from ..utils.theme import register_theme
+
+        # clear out all data to make sure it's up to date (e.g. after
+        # plugin was disabled or unregistered)
+        plugin_manager._theme_data.clear()
+        plugin_manager.discover_themes()
```

Comment:
discover_themes() calls `hook.napari_provide_theme.call_historic` (which only needs to be called once) but this is connected to an event that will trigger this every time a new plugin is registered... is the historic hook still necessary?  (a historic hook is usually only ever called once, and you provide a callback here, `register_theme_colors` that will be called _for you_ each time the plugin manager detects a new plugin that provides the hook)

Reply:
This was actually not necessary so I've removed it. Thanks!